sklearn/decomposition/tests/test_online_lda.py

```diff
@@ -439,3 +440,48 @@ def test_lda_feature_names_out():
     assert_array_equal(
         [f"latentdirichletallocation{i}" for i in range(n_components)], names
     )
+
+
+@pytest.mark.parametrize("learning_method", ("batch", "online"))
+@pytest.mark.parametrize(
+    "train_data_type, expected_type",
+    (
+        (np.float32, np.float32),
+        (np.float64, np.float64),
+        (np.int32, np.float64),
+        (np.int64, np.float64),
+    ),
+)
+def test_lda_dtype_match(learning_method, train_data_type, expected_type):
+    # Verify output matrix dtype
+    dataset_X, _ = make_multilabel_classification(random_state=0)
+
+    lda = LatentDirichletAllocation(
+        n_components=5, random_state=0, learning_method=learning_method
+    )
+    lda.fit(dataset_X.astype(train_data_type))
+    # regardless of input data, transformed data type is determined by trained data
+    for input_data_type in (np.float32, np.float64, np.int32, np.int64):
+        transformed = lda.transform(dataset_X[-2:].astype(input_data_type))
+        assert transformed.dtype == expected_type
+
```

Comment:
```suggestion
        assert lda.components_.dtype == expected_type

```

Reply:
Thank you for pointing out. 
I have added test for `components_` for dtype matching and numerical consistency